libtcmu.c

```diff
@@ -177,8 +177,8 @@ static int reconfig_device(struct tcmulib_context *ctx, char *dev_name,
 
 	ret = dev->handler->reconfig(dev, &cfg);
 	if (ret < 0) {
-		tcmu_dev_err(dev, "Handler reconfig failed with error %d.\n",
-			     ret);
+		tcmu_dev_dbg(dev, "Handler reconfig failed with error %s.\n",
+		             strerror(-ret));
```

Comment:
Is 100% of the annoying logging due to not supported errors (-95)? If so then I think you want to change the tcmu_dev_err at the begging of the function that checks for if a reconfig callout is set.

Reply:
No, as mentioned above, we also possible will have -EINVAL errno.

But even the reconfig callout is support, it will be like:

```
690 static int tcmu_glfs_reconfig(struct tcmu_device *dev,
691                               struct tcmulib_cfg_info *cfg)       
692 {  
693         struct glfs_state *gfsp = tcmur_dev_get_private(dev); 
694         struct stat st;        
695         int ret = -EIO;        
696    
697         switch (cfg->type) {   
698         case TCMULIB_CFG_DEV_SIZE:
699                 ret = glfs_lstat(gfsp->fs, gfsp->hosts->path, &st);                                                                                                      
700                 if (ret) {     
701                         tcmu_dev_warn(dev, "glfs_lstat failed: %m\n");
702                         tcmu_notify_conn_lost(dev);         
703    
704                         /* Let the targetcli command return success */
705                         ret = 0;
706                 } else if (st.st_size != cfg->data.dev_size) {
707                         tcmu_dev_err(dev,                   
708                                      "device size and backing size disagree: device %"PRId64" backing %lld\n",
709                                      cfg->data.dev_size, (long long) st.st_size);
710                         ret = -EINVAL;                      
711                 }
712                 return ret;    
713         case TCMULIB_CFG_DEV_CFGSTR:        
714         case TCMULIB_CFG_WRITE_CACHE:       
715         default:
716                 return -EOPNOTSUPP;
717         }
718 }  
```

Here the most cases we hit this annoying logging is due to TCMULIB_CFG_WRITE_CACHE no support.

Reply:
I think we are talking about the same thing and also different things :)

I was just asking about the really annoying messages. The ones due to target_core_user exporting all attrs even if tcmu-runner handlers do not support them, then rtslib always trying to init them during device creation. For those cases we get the op not supported (-95 errors) here:

https://github.com/open-iscsi/tcmu-runner/blob/b1145f595d960b654295779e3a6429506e202490/libtcmu.c#L156

and

https://github.com/open-iscsi/tcmu-runner/blob/b1145f595d960b654295779e3a6429506e202490/libtcmu.c#L180

pkalaver fixed the second one which glfs hits, but did not fix the first one which can be hit by other handlers.

Reply:
Yeah, make sense to me.

Currently we are only hitting the second one.